src/Mod/PartDesign/Gui/ReferenceSelection.cpp

```diff
@@ -86,13 +92,19 @@ bool ReferenceSelection::allow(App::Document* pDoc, App::DocumentObject* pObj, c
             return false;
     }
 #endif
-    // Handle selection of geometry elements
-    if (Base::Tools::isNullOrEmpty(sSubName))
+    if (!sSubName || sSubName[0] == '\0'){

```

Comment:
isNullOrEmpty was recently added to avoid these manual checks.
Is there any reason why you want to use the manual way?